lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -445,61 +406,139 @@ function Get-CharmServices {
             )
         }
     }
-
+    $s2dContextGen = @{
+        "generator" = (Get-Item "function:Get-S2DCSVContext").ScriptBlock
+        "relation" = "csv"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'csv')) {
+        $s2dContextGen['mandatory'] = $true
+    }
+    $freeRdpContextGen = @{
+        "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+        "relation" = "free-rdp"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'free-rdp')) {
+        $freeRdpContextGen['mandatory'] = $true
```

Comment:
Nothing stopping you from adding a status message saying: "Incomplete relation: s2d". That should be explicit enough. If you really want to add s2d as a special case, when that relation is fulfilled, you can even set a status message saying: "Unit is active and clustered"

I find this is a much better compromise then violating the architectural cleanliness of the code. I am even open to the idea of rethinking the "mandatory/optional" flag for the relationships. If you find a better solution for that, I am all for it. But I do not want to see mandatory options becoming optional on the fly, and vice versa.